AsyncDisplayKit/ASTableView.mm

```diff
@@ -455,6 +455,14 @@ - (void)reloadRowsAtIndexPaths:(NSArray *)indexPaths withRowAnimation:(UITableVi
   [_dataController reloadRowsAtIndexPaths:indexPaths withAnimationOptions:animation];
 }
 
+- (void)relayoutRowAtIndexPath:(NSIndexPath *)indexPath withRowAnimation:(UITableViewRowAnimation)animation
+{
+  ASDisplayNodeAssertMainThread();
+  ASCellNode *node = [self nodeForRowAtIndexPath:indexPath];
+  [node setNeedsLayout];
```

Comment:
Thanks for adding that animation, @nguyenhuy!  I think it is safe to rely on these behaviors you described (very helpful for me to understand).  The main action would be to update the header documentation to make an API guarantee that we intend the reloadRow/Item methods to be callable within an animation block.

I'm going to land this as-is, so take your time and do a small followup when convenient.